Source/DafnyDriver/CSVTestLogger.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace Microsoft.Dafny {
+  public class CSVTestLogger : ITestLoggerWithParameters {
+
+    private readonly ConcurrentBag<TestResult> results = new();
+    private TextWriter writer;
+
+    public void Initialize(TestLoggerEvents events, string testRunDirectory) {
+    }
+
+    public void Initialize(TestLoggerEvents events, Dictionary<string, string> parameters) {
+      events.TestResult += TestResultHandler;
+      events.TestRunComplete += TestRunCompleteHandler;
+      if (parameters.TryGetValue("LogFileName", out string filename)) {
+        writer = filename == "-" ? Console.Out : new StreamWriter(filename);
+      } else {
+        // Auto-generate a file name if none is specified. This uses a
+        // similar approach to the TRX logger, but with simpler logic.
```

Comment:
Appreciate the effort to trim the scope, but unfortunately I think we actually need at least some of the logic to avoid collisions. Otherwise if you invoke `dafny` in parallel (which lots of Makefiles will do) you could actually collide on the same second relatively easily.

Reply:
Fortunately, that logic isn't the most complex part of the TRX implementation, and is pretty easy to replicate (or approximate, with what I think is perhaps a slightly better approach).